core/src/block.rs

```diff
@@ -730,7 +730,7 @@ impl VersionedCommittedBlock {
             transactions,
             event_recommendations,
             ..
-        } = self.into_v1();
+        } = self.clone().into_v1();
```

Comment:
This clone is copying a lot of data. I would prefer if you cloned the only moved value instead.